hopp/simulation/technologies/resource/solar_resource.py

```diff
@@ -38,37 +31,59 @@ def __init__(
         path_resource: Union[str, Path] = ROOT_DIR / "simulation" / "resource_files", 
         filepath: Union[str, Path] ="", 
         use_api: bool = False,
+        resource_data: Optional[dict] = None,
         **kwargs
     ):
+        """Resource class to download solar resource data using API call or set with preloaded dictionary
+
+        Args:
+            lat (float): latitude corresponding to location for solar resource data
+            lon (float): longitude corresponding to location for solar resource data
+            year (int): year for resource data. must be between 1998 and 2022
+            path_resource (Union[str, Path], optional): filepath to resource_files directory. Defaults to ROOT_DIR/"simulation"/"resource_files".
+            filepath (Union[str, Path], optional): full filepath to solar resource data file. Defaults to "".
+            use_api (bool, optional): Make an API call even if there's an existing file. Defaults to False.
+            resource_data (Optional[dict], optional): dictionary of preloaded and formatted solar resource data. Defaults to None.
+            kwargs: extra kwargs
+        """
         super().__init__(lat, lon, year)
 
-        if os.path.isdir(path_resource):
-            self.path_resource = path_resource
+        # if resource_data is input as a dictionary then set_data
+        if isinstance(resource_data,dict):
+            self.data = resource_data
+        # if resource_data is not provided, download or load resource data
+        else:
```

Comment:
I would reduce the control flow by adding an empty `return` above this, and removing the `else` control block so that the remainder of the code is at the same indent level as the first line of the method. In general it's good practice to simplify the control flow as it reduces the complexity of the code. Here, that translates to having a special carve out for the data already existing, and no further work being required.